1_execise.js

```diff
@@ -18,18 +21,58 @@
 
 
 
+const numPet= data.students.map(pet=>{
+  let numOfPet = pet.pets.length;
+  let studentName =pet.studentName;
+  return {studentName,numOfPet}
+})
+
+console.log(numPet);
 /**
  * count of each student's pet
  * show in result: [{studentName: "A", numOfPet: 2}, {studentName: "B", numOfPet: 1}]
  */
 
 
+ const imgStd = data.students.map(std=>{
+  let studentName =std.studentName;
+  let imageOj = data.imageOfStudent.find(img=>img.studentId==std.studentId)
+  let image = imageOj.imgPath
+  return {studentName,image}
+ })
+console.log(imgStd)
+
  /**
  * get image of student 
  * result: [{studentName: "A" , image:"public/img/student1.jpg"}]
  */
+console.log("-----------------------------")
+ const addPetType = data.students.map(All=>{
```

Comment:
suggest, in the future should declare a variable with meaningful like All => eachStudent